### ISSUE TITLE
feat: Notebook support for viz

### ISSUE THREAD

**Issue by kratsg** [alexander-held/cabinetry#163]
There's many ways to skin the visualizations. I guess one way with notebook support is to provide the magic `_repr_x_` functions like in this snippet

```
class Image(BytesIO):
    def __init__(self, content=None, filename=None):
        self.filename = filename
        self.format = filename.split('.')[-1].lower()
        super(Image, self).__init__(content)

    def save(self, filename=None, mode='wb'):
        filename = filename or self.filename
        nbytes = len(self.getvalue())
        with open(filename, mode) as f:
            f.write(self.read())
        log.info('Written {0:d} bytes to {1:s}'.format(len(self.getvalue()), filename))
        return nbytes

    def _repr_png_(self):
        if self.format == 'png':
            return self.getvalue()

    def _repr_jpeg_(self):
        if self.format in ['jpeg', 'jpg']:
            return self.getvalue()

    def _repr_svg_(self):
        if self.format == 'svg':
            return self.getvalue()

    @classmethod
    def from_response(cls, response):
        filename = re.findall(
            "filename=(.+)", response.headers.get('content-disposition')
        )[0]
        return cls(content=response.content, filename=filename)
```

**Comment by alexander-held** [comment]
Thanks for opening this request. The handling of figures is due for some changes, see also discussions in #113 and #142. I have no experience with how notebooks handle images, would they show a figure when calling a function that returns an instance of `Image` in that above example?

**Comment by kratsg** [comment]
> Thanks for opening this request. The handling of figures is due for some changes, see also discussions in #113 and #142. I have no experience with how notebooks handle images, would they show a figure when calling a function that returns an instance of `Image` in that above example?

Yes.

**Comment by alexander-held** [comment]
The removal of `plt.close(fig)` in `src/cabinetry/contrib/matplotlib_visualize.py` makes the figures appear in notebooks. This function was added in #115 due to warnings appearing when too many figures are kept open. Is there maybe a better way to handle this? For now commenting out the lines should make the figures appear, but that is not a good long term solution.

**Comment by alexander-held** [comment]
#240 will change the default behavior to not immediately close figures after saving, which will make them show up in notebooks again. There should be more figure API changes coming for `cabinetry` 0.3, so keeping this issue open for now.

**Comment by alexander-held** [comment]
@kratsg what is the advantage of the `_repr_x` approach compared to keeping `matplotlib` figures open so they are still around and rendered at the end of a cell execution? #264 will also return figures when calling functions in the `visualize` module, thereby also allowing to render them again.

**Comment by alexander-held** [comment]
Considering this as resolved via #240 and #264. Please feel free to re-open if there are still things missing, or there are possible improvements.
